### PR TITLE
Implement AOS-62 WriteMultipart returns Part

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/aos-dev/go-storage/issues/571.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -37,6 +38,9 @@ func (s *Storage) completeMultipart(ctx context.Context, o *Object, parts []*Par
 	if err != nil {
 		return
 	}
+
+	o.Mode &= ^ModePart
+	o.Mode |= ModeRead
```
Comment (Xuanwo) [inline storage.go:43]: Maybe we can extract to function `Add` && `Del`?

```
o.Mode.Add(ModeRead)
o.Mode.Del(ModePoart)
```

File: storage.go [hunk 2]
```diff
@@ -583,9 +587,15 @@ func (s *Storage) writeMultipart(ctx context.Context, o *Object, r io.Reader, si
...
+	output, err := s.bucket.UploadMultipartWithContext(ctx, o.ID, input)
 	if err != nil {
 		return
 	}
-	return size, nil
+
+	part = &Part{
+		Index: index,
+		Size:  size,
+		ETag:  *output.ETag,
```
Comment (Xuanwo) [inline storage.go:598]: Use `service.StringValue(v.ETag)` here to prevent panic.

File: storage.go (hunk 1 above)
Comment (JinnyYi) [reply storage.go:43]: Should I add function `Add` and `Del` in `go-storage` or just add in utils.go?

Comment (Xuanwo) [reply storage.go:43]: Let's add them in go-storage.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**